### PR TITLE
Add Restart action to Deployment details page

### PR DESCRIPTION
**Pull request by trentcain**
Fixes: #6639

Backend Changes -
1. Added Restartable to TypeMeta. This is set to true for Deployment resources (similar to Scalable).

Frontend Changes -
1. Added new restart detail action component (ActionbarDetailRestartComponent). The component contains a button with the refresh mat icon, clicking the button shows the restart dialog.
2. Added the Restart action component to the Deployment action bar (ScaleDefaultActionbar). The action is only visible if typeMeta.restartable is true. 
Note for reviewer: I wasn't sure if this action bar should be renamed to something like ScaleRestartDefaultActionbar. Please let me know if this is required, or if this new restart detail action component belongs in a different component.
3. Updated the existing list menu component to use typeMeta.restartable to determine if restart should be enabled in the menu.

Screenshot of new component:
Screenshot 2021-12-01 183854

Video of new component:

https://user-images.githubusercontent.com/53938374/144210422-91174721-8ea4-4249-aa97-77f469c6f925.mp4

### CODE REVIEW

**Comment by maciaszczykm** [comment]
/lgtm

Looks solid.

File: src/app/frontend/common/components/actionbar/detailactions/restart/template.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<button mat-icon-button
+        color="accent"
+        class="kd-toolbar-action"
+        i18n-matTooltip
+        matTooltip="Restart resource"
```
Comment (floreks) [inline src/app/frontend/common/components/actionbar/detailactions/restart/template.html:21]: nit: s/Restart resource/Restart

**Review by floreks — CHANGES_REQUESTED** [review]
Just a very minor nit pick :slightly_smiling_face: 

This is a really good idea to move this part and control it on the backend side. Code also looks good. Thanks!

**Comment by trentcain** [comment]
Tooltip updated. Thanks @floreks

**Review by floreks — APPROVED** [review]

**Pull request merged**